internal/access/access_key.go

```diff
@@ -35,16 +35,14 @@ func ListAccessKeys(c *gin.Context, identityID uid.ID, name string, showExpired
 func CreateAccessKey(c *gin.Context, accessKey *models.AccessKey) (body string, err error) {
 	rCtx := GetRequestContext(c)
 
-	if rCtx.Authenticated.AccessKey != nil && !rCtx.Authenticated.AccessKey.Scopes.Includes(models.ScopeAllowCreateAccessKey) {
-		// non-login access keys can not currently create other access keys.
-		return "", fmt.Errorf("%w: cannot use an access key not issued from login to create other access keys", internal.ErrBadRequest)
-	}
+	if err := IsAuthorized(rCtx, models.InfraAdminRole); err != nil {
+		// not an admin
+		if rCtx.Authenticated.AccessKey != nil && !rCtx.Authenticated.AccessKey.Scopes.Includes(models.ScopeAllowCreateAccessKey) {
```

Comment:
Love that we've decided to make this an admin user thing since that's easier for users.

Not a showstopper but this check "bypasses" the `create-key` scope which breaks the integrity of the scope. Instead, could we make it so keys created by admins can include this scope by (maybe by default for now) and both the scope and admin role are required here?